app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -291,31 +302,180 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
     }
 
     /**
-     * Determines whether the cart has either all items available if Manage Stock is yes for requested quantities,
-     * or, if not, those items are eligible for back order.
+     * @param $quoteId
+     *
+     * @return \Mage_Sales_Model_Quote
+     */
+    protected function getQuoteById($quoteId)
+    {
+        /* @var Mage_Sales_Model_Quote $immutableQuote */
+        $immutableQuote = Mage::getModel('sales/quote')
+            ->getCollection()
+            ->addFieldToFilter('entity_id', $quoteId)
+            ->getFirstItem();
+
+        if($this->allowDisabledSKUOrders()) {
+            $immutableQuote->setIsSuperMode(true); // Allow an order to be created even if it has disabled products
+        }
+
+        return $immutableQuote;
+    }
+
+    /**
+     * @return boolean
+     */
+    protected function allowDisabledSKUOrders()
+    {
+        return Mage::getStoreConfigFlag('payment/boltpay/allow_disabled_sku_orders');
+    }
+
+    /**
+     * @return boolean
+     */
+    protected function allowOutOfStockOrders()
+    {
+        return Mage::getStoreConfigFlag('payment/boltpay/allow_out_of_stock_orders');
+    }
+
+    /**
+     * @param Mage_Sales_Model_Quote $quote
      *
-     * @var Mage_Sales_Model_Quote $quote   The quote that defines the cart
+     * @return void
+     */
+    protected function validateProducts(Mage_Sales_Model_Quote $quote)
+    {
+        $outOfStockSKUs = $this->getOutOfStockSKUs($quote);
+        if ($outOfStockSKUs) {
+            $this->enableOutOfStockOrderToBeCreated();
+            $this->setOrderOnHoldFlag();
+
+            $errorMessage = Mage::helper('boltpay')->__("Product " .
+                join(", ", $outOfStockSKUs) .
+                (count($outOfStockSKUs) == 1 ? " is" : " are") .
+                " out of stock. ");
+
+            $this->appendOrderOnHoldMessage($errorMessage);
+        }
+
+        $disabledSKUs = $this->getDisabledSKUs($quote);
+        if ($disabledSKUs) {
+            $this->setOrderOnHoldFlag();
+
+            $errorMessage = Mage::helper('boltpay')->__("Product " .
+                join(", ", $disabledSKUs) .
+                (count($disabledSKUs) == 1 ? " is" : " are") .
+                " disabled. ");
+
+            $this->appendOrderOnHoldMessage($errorMessage);
+        }
+
+        if ($this->isOrderOnHoldFlagSet()) {
+            $invalidSKUs = array_unique(array_merge($outOfStockSKUs, $outOfStockSKUs));
```

Comment:
This code checks to see if the order went on hold. This can happen because an item is out of stock, or because an item was disabled (not out of stock but it's no longer enabled on their site)